web/src/components/datasets/DatasetInfo.tsx

```diff
@@ -21,38 +21,44 @@ interface DatasetInfoProps {
 const DatasetInfo: FunctionComponent<DatasetInfoProps> = props => {
   const { datasetFields, facets, run } = props
 
+  let datasetFieldsBox;
+  
+
   if (datasetFields.length === 0) {
-    return <MqEmpty title={'No Fields'} body={'Try adding dataset fields.'} />
+    datasetFieldsBox = <MqEmpty title={'No Fields'} body={'Try adding dataset fields.'} />
```

Comment:
Setting a variable like this is a little misleading. 

One nice way to render this out without the overall conditional is to use a ternary to avoid the variable assignment.

```tsx
{datasetFields.length === 0 && <MqEmpty title={'No Fields'} body={'Try adding dataset fields.'} />}
```